### PR TITLE
clkinfomsg: clockinfo for displaying messages count

### PR DESCRIPTION
**Pull request by kamilkrz**
Read and unread. This is proof of concept but mostly working.
Only nice to have would be handling messages that was read on phone (I believe if keep msgs is not set in android, messages read on phone trigger some kind of event for removing it from watch)

Nothing fancy

### CODE REVIEW

File: apps/clkinfomsg/clkinfo.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  }
+
+  var debug = function(o) {
+    //console.log(o);
+  }
+  var msgUnread;
+  var msgAll;
+  var msgs = require("messages");
+
+  var getAllMGS = function() {
```
Comment (bobrippling) [inline apps/clkinfomsg/clkinfo.js:17]: Minor typo - should this be `getAllMSGs` ?

File: apps/clkinfomsg/clkinfo.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    if (msgUnread === undefined) {
+      debug("msgUnread is undefined");
+      msgUnread = msgs.getMessages().filter(m => m.new && !['call', 'map', 'music'].includes(m.id)).map(m => m.src).length;
+    }
+    return msgUnread; 
+  }
+
+  var msgCounter = function(type, msg) {
+    var msgsNow = msgs.getMessages(msg);
+    msgUnread = msgsNow.filter(m => m.new && !['call', 'map', 'music'].includes(m.id)).map(m => m.src).length;
```
Comment (bobrippling) [inline apps/clkinfomsg/clkinfo.js:36]: Do we need the `.map()` if we then just find the length of the array? (Similarly for a few other instances)

File: apps/clkinfomsg/clkinfo.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    var msgsNow = msgs.getMessages(msg);
+    msgUnread = msgsNow.filter(m => m.new && !['call', 'map', 'music'].includes(m.id)).map(m => m.src).length; 
+    if (msgUnread === undefined) {
+      msgUnread = "?";
+    }
+    msgAll = msgsNow.filter(m => !['call', 'map', 'music'].includes(m.id)).map(m => m.src).length; 
+    if (msgAll === undefined) {
+      msgAll = "?";
+    }
+    info.items[0].emit("redraw");
```
Comment (bobrippling) [inline apps/clkinfomsg/clkinfo.js:44]: What happens here if `items[1]` is being displayed?

File: apps/clkinfomsg/clkinfo.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  }
+
+  var msgCounter = function(type, msg) {
+    var msgsNow = msgs.getMessages(msg);
+    msgUnread = msgsNow.filter(m => m.new && !['call', 'map', 'music'].includes(m.id)).map(m => m.src).length; 
+    if (msgUnread === undefined) {
+      msgUnread = "?";
+    }
+    msgAll = msgsNow.filter(m => !['call', 'map', 'music'].includes(m.id)).map(m => m.src).length; 
+    if (msgAll === undefined) {
```
Comment (bobrippling) [inline apps/clkinfomsg/clkinfo.js:41]: I don't think the length can be `undefined`, perhaps you mean `0`?

**Review by bobrippling — COMMENTED** [review]
Thanks for the app! Had a few questions :)

File: apps/clkinfomsg/clkinfo.js (hunk 1 above)
Comment (kamilkrz) [reply apps/clkinfomsg/clkinfo.js:17]: Good catch - fixing

File: apps/clkinfomsg/clkinfo.js (hunk 2 above)
Comment (kamilkrz) [reply apps/clkinfomsg/clkinfo.js:36]: Oh totally, I was testing different approach.

File: apps/clkinfomsg/clkinfo.js (hunk 4 above)
Comment (kamilkrz) [reply apps/clkinfomsg/clkinfo.js:41]: Yup, leftover from old approach - will fix and test.

File: apps/clkinfomsg/clkinfo.js (hunk 3 above)
Comment (kamilkrz) [reply apps/clkinfomsg/clkinfo.js:44]: I need some guidance - is there a chance to have context here about current CI is being shown? I don't like idea of refreshing all CI modules but in my opinion its best place to do it here.

Comment (bobrippling) [reply apps/clkinfomsg/clkinfo.js:39]: I'll merge, but feel free to open another PR if you want to tweak it :)

**Pull request merged**